Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -47,21 +53,86 @@ public final class HTTPClientCopyingDelegate: HTTPClientResponseDelegate {
+    ) throws -> NIOClientTCPBootstrap {
+        // if there is a proxy don't create TLS provider as it will be added at a later point
+        if configuration.proxy != nil {
+            return NIOClientTCPBootstrap(self, tls: NIOInsecureNoTLS())
+        } else {
+            let tlsConfiguration = configuration.tlsConfiguration ?? TLSConfiguration.forClient()
+            let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
+            let hostname = (!requiresTLS || host.isIPAddress) ? nil : host
+            let tlsProvider = try NIOSSLClientTLSProvider<ClientBootstrap>(context: sslContext, serverHostname: hostname)
+            return NIOClientTCPBootstrap(self, tls: tlsProvider)
+        }
+    }
+}
 
+extension NIOClientTCPBootstrap {
+    
+    /// create a TCP Bootstrap based off what type of `EventLoop` has been passed to the function.
+    fileprivate static func makeBootstrap(
+        on eventLoop: EventLoop,
+        host: String,
+        requiresTLS: Bool,
+        configuration: HTTPClient.Configuration
+    ) throws -> NIOClientTCPBootstrap {
+        let bootstrap: NIOClientTCPBootstrap
+        #if canImport(Network)
+        // if eventLoop is compatible with NIOTransportServices create a NIOTSConnectionBootstrap
+        if #available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *),
+            let tsBootstrap = NIOTSConnectionBootstrap(validatingGroup: eventLoop) {
+            let tlsConfiguration = configuration.tlsConfiguration ?? TLSConfiguration.forClient()
+            // if there is a proxy don't create TLS provider as it will be added at a later point
+            if configuration.proxy != nil {
+                bootstrap = NIOClientTCPBootstrap(tsBootstrap, tls: NIOInsecureNoTLS())
+            } else {
+                // create NIOClientTCPBootstrap with NIOTS TLS provider
+                let parameters = tlsConfiguration.getNWProtocolTLSOptions()
+                let tlsProvider = NIOTSClientTLSProvider(tlsOptions: parameters)
+                bootstrap = NIOClientTCPBootstrap(tsBootstrap, tls: tlsProvider)
+            }
+        } else if let clientBootstrap = ClientBootstrap(validatingGroup: eventLoop) {
+            bootstrap = try clientBootstrap.makeClientTCPBootstrap(host: host, requiresTLS: requiresTLS, configuration: configuration)
+        } else {
+            preconditionFailure("Cannot create bootstrap for the supplied EventLoop")
+        }
+        #else
+        if let clientBootstrap = ClientBootstrap(validatingGroup: eventLoop) {
+            bootstrap = try clientBootstrap.makeClientTCPBootstrap(host: host, requiresTLS: requiresTLS, configuration: configuration)
+        } else {
+            preconditionFailure("Cannot create bootstrap for the supplied EventLoop")
+        }
+        #endif
+        
+        // don't enable TLS if we have a proxy, this will be enabled later on
+        if requiresTLS, configuration.proxy == nil {
```

Comment:
```suggestion
        if requiresTLS && configuration.proxy == nil {
```

Only use comma if conditional binding is involved.